casanovo/denovo/model.py

```
@@ -142,11 +142,13 @@ def __init__(
         # unrecognized arguments, including deprecated ones. Remove the
         # deprecated ones.
         for k in config._config_deprecated:
-            kwargs.pop(k, None)
-            warnings.warn(
-                f"Deprecated hyperparameter '{k}' removed from the model.",
-                DeprecationWarning,
-            )
+            if k in kwargs:
+                kwargs.pop(k)
+                warnings.warn(
```

Ruff (0.15.6)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)